presto-spi/src/main/java/com/facebook/presto/spi/type/UnscaledDecimal128Arithmetic.java

```diff
@@ -698,6 +707,9 @@ private static void setNegative(Slice decimal, boolean negative)
 
     private static void setNegativeInt(Slice decimal, int v3, boolean negative)
     {
+        if (v3 < 0) {
```

Comment:
Is that required? Operators that always pass valid v3 will pay additional check cost.